users/permissions.py

```diff
@@ -0,0 +1,12 @@
+"""Decorators for checking permissions."""
+from django.contrib.auth.decorators import login_required, user_passes_test
+
+
+def user_is_logged_in_and_active(function):
+    user_is_active = user_passes_test(lambda u: u.is_active, login_url="denied/")
+    return login_required(user_is_active(function), login_url="/")
+
+
+def user_is_staff(function):
+    staff_user = user_passes_test(lambda u: u.is_staff, login_url="denied/")
+    return login_required(staff_user(function), login_url="/")
```

Comment:
Doesn't the `user_passes_test` already check if the user is logged in?

Reply:
If there is no logged in user, then request.user is an AnonymousUser object who fails every test. `user_passes_test` would just give `False` without checking for log in.h

Reply:
Hmm, both sides make sense. If not logged in, technically, the user should be shown the access denied page, but it would be better UX if they are redirected to login.

Reply:
You could also chain decorators IIRC. Like so:
```python
@user_passes_test(...)
@login_required(...)
def my_view():...
```

Reply:
https://stackoverflow.com/questions/8715642/multiple-decorators-for-a-view-in-django-execution-order

Reply:
NVM, I guess the current way follows best practices.